docs/internal/capabilities.md

```diff
@@ -0,0 +1,210 @@
+...
+var bar : Foo & readonly = foo; // OK: a read-only reference to a mutable object
+...
+var baz : Foo & mut = bar; // ERROR: cannot convert readonly to mutable
+```
+
+**Immutable objects** can also alias each other, but since none of those references can write through them, once created (frozen), an immutable region can only be read from.
+
+```ts
+var foo : Foo & imm = freeze(Foo::create(10));
+print(foo.value); // prints "10"
+...
+var bar = foo;
+print(bar.value); // prints "10"
+...
+var ro : Foo & readonly = bar; // OK: imm is read-only anyway, so a noop
+var imm : Foo & imm = ro; // OK? also a noop but there's a sub-type relationship that I'm not sure is there
+```
+
+### Destructive reads
+
+Some reads invalidate the previous reference on assignment.
+By design, `iso` references have a destructive read, since the program cannot have two `iso` references to the same region.
+But behaviour captures also destroy the previous reference to avoid race conditions.
+
+Example:
+```ts
+var foo : Foo & mut = Foo;
+// Creates an anonymous asynchronous behaviour
+// Captures foo into a field of the lambda, say "_field0".
+// That capture is a destructive read.
+when ()
+{
+    // Uses foo here, which in the lambda is the same as _field0.doSomething();
+    // Because the original foo is no longer valid.
+    foo.doSomething();
+} // Schedules the behaviour and return -before- the execution finishes
+...
+// If this was allowed, there could be a race condition with _field0
+foo.somethingElse(); // ERROR: foo was captured into a new reference and this one is now invalid
+```
+
+## Capability Conversions
+
+Verona does not permit implicit type conversions and that also applies to capabilities.
+Using a variable with one capability in place of another will not coerce the compiler to convert capabilities, unless there is a clear sub-type relationship or conversion function (ex. `freeze`).
+
+### Sub-typing
+
+Sub-typing is the type relationship where a sub-type can be safely used in place of its super type, for example, as function arguments or return values.
+Sub-types can, therefore, be passed as arguments or directly assigned to variables that have been declared as their super-types.
+
+Example:
+```ts
+interface Super {}
+class Sub : Super {}
+function(arg : Super) { ... }
+...
+a : Sub = Sub;
+function(a); // OK, as Sub is a sub-type of Super
+```
+
+The notation is `Sub <: Super` that means `Sub` is a sub-type if `Super`.
+
+### Allowed Conversions
+
+These are the following capability sub-type relationships in Verona:
+* `iso <: mut`: Passing an `iso` in a (behaviour-local) place expecting a `mut` will _view_ the type as a mutable reference, because `iso` is strictly stringer than `mut`.
+* `mut <: readonly`: Passing a `mut` in a place expecting a `readonly` will _view_ the type as read-only and not allow writes to it.
+* `imm <: readonly`:  Similar reasoning, but here it's a no-op, because the object was globally immutable to begin with.
```

Comment:
```suggestion
* `imm <: readonly`:  An `imm` capability guarantees global immutability, `readonly` guarantees local immutability which is a strictly weaker property that is trivially satisfied by the stronger property.
```

I don't like the 'no-op' framing because that implies that the other conversions are non-trivial operations.